caffe2/core/tensor.cc

```diff
@@ -124,6 +124,34 @@ Tensor empty(at::IntList dims, at::TensorOptions options) {
   return tensor;
 }
 
+void reinitialize_(
+    Tensor* tensor,
+    at::IntList dims,
+    at::TensorOptions options) {
+  CAFFE_ENFORCE(options.device_opt() != c10::nullopt);
```

Comment:
OOC, why not use the defaulting? (`options.device()`)

Reply:
because of the specific use case we are expecting(for actual initialization), I think it might be better to enforce users to provide device type? I don't have strong opinion on this tho, we may say when we use it in c2 code base we'll always provide it as well.

Reply:
given that we want to use in C2 ops for now - it's better to make it explicit indeed